p2p/agent.go

```diff
@@ -192,7 +236,13 @@ func NewAgent(cfg Config, chainID uint32, genesisHash hash.Hash256, broadcastHan
 		unicastInboundAsyncHandler: unicastHandler,
 		reconnectTimeout:           cfg.ReconnectInterval,
 		qosMetrics:                 NewQoS(time.Now(), 2*cfg.ReconnectInterval),
+		networks:                   make(map[string]struct{}),
 	}
+	a.networks[BlockNetwork] = struct{}{}
+	for _, opt := range opts {
+		opt(a)
+	}
```

Comment:
if len(a.networks) == 0 { error out }

Reply:
node can not join any subnets in p2p level